### DISCUSSION TITLE
Welcome to btcd Discussions!

### DISCUSSION THREAD

**Discussion by Roasbeef** [btcsuite/btcd#1950] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by jcvernaleo** [comment]
This looks like it should be really useful.  irc kind of lost the critical mass for being good enough for discussion sadly.

**Comment by Roasbeef** [comment]
Yeah, we also get a lot of trouble shooting or library usage issues that
can be converted into discussions with a single button.

**Comment by souvikhaldar** [comment]
IRC is still my go to way of communication!

**Comment by mlitanin** [comment]
is there any way to know how many of this implementation of the full nodes currently running in production?
lets say today 15k activate nodes on average(https://bitnodes.io/), how many of them are running on btcd ?

**Reply by guggero** [reply]
https://bitnodes.io/nodes/?q=btcwire

**Reply by mlitanin** [reply]
i see there are about 23, and many of them aren't live. is there particular reason this implementation is not being widely used?

**Comment by souvikhaldar** [comment]
I’m running btcd as well as bitcoind but I see IBD on btcd is painfully slow and no way to speed up. Plus I don’t think electrum/fulcrum/etc runs on top of btcd!but if you write go compatibility is better interacting with btcd than bitcoind.On 08-Dec-2023, at 12:09 AM, maksym p. ***@***.***> wrote:﻿
is there any way to know how many of this implementation of the full nodes currently running in production?
lets say today 15k activate nodes on average(https://bitnodes.io/), how many of them are running on btcd ?

**Comment by mlitanin** [comment]
i see, yes i agree IBD is painfully slow. been running it for few days , still on year 2017 ))
why there is no way to speed it up? another question - why do we pull it block by block, why not make some giant archive with "everything before 2023" which we can download as one file and then run the validation locally over all the blocks and make sure the final hash matches the consensus one.
sorry if my question if dumb, i am just starting learning the technicalities of the protocol

**Comment by jcvernaleo** [comment]
A long time ago there was a torrent of the bitcoin block data up to a certain point.  I think what happened was the combination of maintaining it, hosting it, and trusting it combined to be more of a problem than anyone wanted to deal with so it sort of stopped.

There may have been some other details there but it was a long time ago so I'm a little fuzzy on it.

**Reply by mlitanin** [reply]
do you think it worths while to implement this logic in btcd so at least between btcd nodes we can have this option for IBD, its not official protocol command, but implemented to just improve the bootstrap of btcd node from scratch

**Reply by jcvernaleo** [reply]
My best guess is that having a second path for IBD is not worth the added complexity.

**Comment by Roasbeef** [comment]
FWIW you can make distinct questions on here for Discussion, re IBD speed:
  * With https://github.com/btcsuite/btcd/pull/1955 I can sync from a local `bitcoind` node in 16 hours. 
  * If you add some of the other GC optimization PRs, that gets down closer to 12 hours. 
  * The missing rung is parallel block download, without that, if you have one slow peer, then the entire sync is bogged down, and the network bandwidth isn't fully saturated.

**Reply by mlitanin** [reply]
as i can see from this PR, there are few things that aren't finalized to merge it and some known issues.
there is mainly 1 implementation of the protocol and vast majority of the nodes running it. its known to be an issue and
i wonder what is the second best implementation and if this one (on go, btcd) can be such. 
if the only downside is the IBD may be there is a workaround (like be able to download archive of all chain to date) or if something can be done to improve it within btcd (like in PR above given the issues are fixed) and have a solid alternative implementation of the protocol in go. Today there are like 20 nodes running it, which is 0.1% (average number of nodes online is 15k).

Unfortunately i am not good enough at programming to really contribute to those issues with PR or suggest other material improvements, but I am trying to understand generally where do we stand in terms of alternative implementation of the protocol.
